### PR TITLE
Disable admin in wizard role picker for non keybase users

### PR DESCRIPTION
**Pull request by jakob223**
(draft because this is waiting on a web of PRs to be merged before adding the same thing to the generate invite link modal)

cc @keybase/y2ksquad

### CODE REVIEW

File: shared/teams/add-members-wizard/confirm.tsx [hunk 1]
```diff
@@ -30,6 +44,14 @@ const AddMembersConfirm = () => {
       addingMembers.findIndex(member => !member.assertion.endsWith('@email')) === -1,
     [addingMembers]
   )
+  const emailsAndPhones = React.useMemo(
```
Comment (buoyad) [inline shared/teams/add-members-wizard/confirm.tsx:47]: We should check for any non keybase user, twitter assertions and such

File: shared/teams/add-members-wizard/confirm.tsx [hunk 2]
```diff
@@ -17,10 +17,24 @@ import {ModalTitle, ChannelsWidget} from '../common'
 import {pluralize} from '../../util/string'
 import logger from '../../logger'
 
+type DisabledRoles = React.ComponentProps<typeof FloatingRolePicker>['disabledRoles']
+const disabledRolesForEmailsPhones = {
+  owner: null,
```
Comment (buoyad) [inline shared/teams/add-members-wizard/confirm.tsx:22]: Designs have this not showing for the group setting at all, not sure that really makes sense though.

Comment (jakob223) [reply shared/teams/add-members-wizard/confirm.tsx:22]: Even when you're adding multiple keybase users? I've wanted to add them as owners before though :|

File: shared/teams/add-members-wizard/confirm.tsx (hunk 1 above)
Comment (jakob223) [reply shared/teams/add-members-wizard/confirm.tsx:47]: I forgot you could get other assertions here

File: shared/teams/team/invites/generate-link.tsx [hunk 3]
```diff
@@ -179,7 +179,7 @@ const GenerateLinkModal = (props: Props) => {
   const rolePickerProps = {
     disabledReasonsForRolePicker: {
       admin: `Users can't join open teams as admins.`,
```
Comment (jakob223) [inline shared/teams/team/invites/generate-link.tsx:181]: fix

Comment (jakob223) [reply shared/teams/team/invites/generate-link.tsx:181]: also fix corresponding thing in new team info

**Comment by buoyad** [comment]
@jakob223 would be good to get this in with discussed reducer changes

**Comment by jakob223** [comment]
Made those changes this morning but forgot to push them.

File: shared/constants/teams.tsx [hunk 4]
```diff
@@ -928,7 +928,14 @@ export const dedupAddingMembeers = (
   const existing = [..._existing]
   for (const toAdd of toAdds) {
     if (!existing.find(m => m.assertion === toAdd.assertion)) {
-      existing.unshift(toAdd)
+      const role =
+        (toAdd.role === 'admin' || toAdd.role === 'owner') && toAdd.assertion.includes('@')
```
Comment (buoyad) [inline shared/constants/teams.tsx:932]: :sheep: kinda bugged by a dedup function also doing this, I think it's fine to rely on the thing inserting these to not use an invalid role?

Comment (jakob223) [reply shared/constants/teams.tsx:932]: Yeah you're right this isn't the right place for this.

**Pull request merged**